controllers/user/resetPassword.js

```diff
@@ -49,22 +47,14 @@ const resetPassword = async (req, res, next) => {
 
     req.body.password = bcrypt.hashSync(req.body.password, salt);
 
-    const user = await models.User.findOneAndUpdate(
+    delete req.body.currentPassword;
+    await models.User.findOneAndUpdate(
       { username: req.user.username },
       req.body
     )
       .select(['-password'])
       .lean();
 
-    if (!user) {
-      return next(
-        createError({
-          message: 'Please login, to reset password',
-          status: UNAUTHORIZED
-        })
-      );
-    }
-
```

Comment:
Why did you remove line 59 - 66?

Reply:
For one, because a missing token is already detected by the verifyToken middle ware. 
Secondly there is already the same check done in line 24-31